Firestore/src/Query.php

```diff
@@ -313,6 +315,19 @@ public function where($fieldPath, $operator, $value)
             ));
         }
 
+        if ($escapedPathString === self::DOCUMENT_ID &&
```

Comment:
This is a rather complex conditional, WDYT about breaking this down into something like the following?

```php
if ($escapedPathString === self::DOCUMENT_ID && !$this->isValidDocument($value, $basePath)) {
    throw new \InvalidArgumentException(
        'When filtering on document ID, value must be a document reference or valid document name.'
    );
}

private function isValidDocument($value, $basePath)
{
    if ($value instanceof DocumentReference || $value instanceof DocumentSnapshot) {
        return true;
    }

    if (is_string($value) && $this->isDocument($this->childPath($basePath, $value))) {
        return true;
    }

    return false;
}
```